### DISCUSSION TITLE
Request body is not preloaded with example data when the schema has nested objects

### DISCUSSION THREAD

**Discussion by sankarshan-nous** [Azure/api-management-developer-portal#2626] [Feature request]
The sample request body is not preloaded for few post/put requests which are having nested schema references. I tried by adding sample request under design menu of the endpoint but that does not reflect in the APIM.  

![APIM-Post](https://github.com/Azure/api-management-developer-portal/assets/109160187/15015e44-128f-4587-a174-1eca9487b21a)

Below is the example request schema which is used for the post request. I am expecting the APIM to generate example request object in the body of the request.

"RequestDto": {
  "type": "object",
  "properties": {
    "email": {
      "type": "string",
      "nullable": true
    },
    "firstName": {
      "type": "string",
      "nullable": true
    },
    "lastName": {
      "type": "string",
      "nullable": true
    },
    "mobileNumber": {
      "type": "string",
      "nullable": true
    },
    "startDate": {
      "type": "string",
      "format": "date-time",
      "nullable": true
    },
    "endDate": {
      "type": "string",
      "format": "date-time",
      "nullable": true
    },
    "inviteCommunicationMethod": {
      "$ref": "#/components/schemas/CommunicationMethod"
    },
    "userId": {
      "type": "string",
      "format": "uuid",
      "nullable": true
    },
    "companyId": {
      "type": "string",
      "format": "uuid",
      "nullable": true
    },
    "business": {
      "$ref": "#/components/schemas/Business"
    },
    "requestType": {
      "$ref": "#/components/schemas/RequestType"
    },
    "personalDataDto": {
      "$ref": "#/components/schemas/PersonalDataDto"
    }
  },
  "additionalProperties": false
},

**Comment by malincrist** [comment]
hello @sankarshan-nous , thanks for reaching out.
I'm not able to reproduce the issue you are describing. 

I added a response with the schema you provided and then generated a sample from Azure Portal:
![image](https://github.com/Azure/api-management-developer-portal/assets/92857141/2e13031c-6d48-464f-ab1d-ecef93952923)

then, this sample is loaded in developer portal
![image](https://github.com/Azure/api-management-developer-portal/assets/92857141/db1b6389-e060-4922-bc3e-de3775371662)

**Comment by sankarshan-nous** [comment]
Oh sorry. I mentioned it wrong. It does generate a sample request when I have already added an example under design. What I am expecting is to give a sample request body to the user when sending a request without adding sample request under design manually. 
The confusion is APIM provides sample request body preloaded for few Api's but not for all. One of the cases I observed was for the schema I shared earlier. Shouldn't it generate automatically under "Try it" section even if it is not generated under design? Because to add the sample request body manually under design will be a huge task for all endpoints/applications.
